scripts/atomic_commit_manager.py

```
@@ -0,0 +1,269 @@
+                # Get commit hash
+                hash_result = subprocess.run([
+                    "git", "rev-parse", "HEAD"
+                ], cwd=worktree_path, capture_output=True, text=True, check=True)
+                return hash_result.stdout.strip()
+            else:
+                # Check if it's because there's nothing to commit
+                if "nothing to commit" in result.stderr:
+                    # Get current HEAD hash
+                    hash_result = subprocess.run([
+                        "git", "rev-parse", "HEAD"
+                    ], cwd=worktree_path, capture_output=True, text=True, check=True)
```

Ruff (0.14.4)

Starting a process with a partial executable path

(S607)